sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -391,6 +394,54 @@ public virtual async Task<string[]> GetPartitionIdsAsync(CancellationToken cance
             return await Connection.GetPartitionPropertiesAsync(partitionId, RetryPolicy, cancellationToken).ConfigureAwait(false);
         }
 
+        /// <summary>
+        ///   A set of information about the state of publishing for a partition, as observed by the <see cref="EventHubProducerClient" />.  This
+        ///   data can always be read, but will only be populated with information relevant to the features which are active for the producer client.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The unique identifier of a partition associated with the Event Hub.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken" /> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>The set of information about the publishing state of the requested partition, within the context of this producer.</returns>
+        ///
+        public virtual async ValueTask<PartitionPublishingProperties> ReadPartitionPublishingPropertiesAsync(string partitionId,
```

Comment:
`ValueTask` is used here because invocation is expected to be synchronous the vast majority of time and for local consistency in API shape with the `CreateBatchAsync` method, which has the same considerations.

Reply:
Reconsidered the approach and refactored to always hold the semaphore to ensure consistency.